lib/node_renderer.rb

```diff
@@ -0,0 +1,94 @@
+require_relative 'dom_parser.rb'
+require 'pry'
+
+
+class NodeRenderer
+
+  attr_reader :tree
+
+  def initialize
+    @dom_parser = DomParser.new
+    @tree = @dom_parser.build_tree("../test.html")
+    @child_count = 0
+    @types_hash = {}
+
+  end
+
+  def render(node=nil)
```

Comment:
In addition to rendering, would be a good idea to save the information in a data structure that you could then re-use. Adding all the info to a hash or array...